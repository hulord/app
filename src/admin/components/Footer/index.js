import React from "react";
import { Layout } from "antd";
import classnames from "classnames";
import styles from "./index.module.less";

/**
 * Renders the global footer
 */
const Footer = props => (
  <Layout.Footer className={classnames(styles.footer, props.className)}>
    Â© monkey admin. All <span className={styles.bugs}>bugs</span> reserved.
  </Layout.Footer>
);

export default Footer;
