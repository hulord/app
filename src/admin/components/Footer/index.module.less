@import "../../theme.less";

.footer {
  text-align: center;
  color: #6272a4;
  font-family: @font-family-brand;
}

.bugs {
  transform: rotate(180deg);
  transition: all 0.5s ease-in-out;
  display: inline-block;
}

.bugs:hover {
  transform: rotate(0);
}
