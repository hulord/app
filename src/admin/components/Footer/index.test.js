import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";

import Footer from "./index";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Footer />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders correctly", () => {
  const tree = renderer.create(<Footer />).toJSON();
  expect(tree).toMatchSnapshot();
});
