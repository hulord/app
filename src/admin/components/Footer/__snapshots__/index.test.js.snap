// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ant-layout-footer"
>
  Â© monkey admin. All 
  <span
    className={undefined}
  >
    bugs
  </span>
   reserved.
</div>
`;
